app/src/sharedTest/java/org/oppia/android/app/devoptions/DeveloperOptionsActivityTest.kt

```diff
@@ -98,10 +141,230 @@ class DeveloperOptionsActivityTest {
   }
 
+  @Test
+  fun testDeveloperOptionsFragment_modifyLessonProgressIsDisplayed() {
+    launch<DeveloperOptionsActivity>(
+      createDeveloperOptionsActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      verifyItemDisplayedOnDeveloperOptionsListItem(
+        itemPosition = 0,
+        targetView = R.id.modify_lesson_progress_text_view
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 0,
+        targetViewId = R.id.mark_chapters_completed_text_view,
+        stringIdToMatch = R.string.developer_options_mark_chapters_completed
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 0,
+        targetViewId = R.id.mark_stories_completed_text_view,
+        stringIdToMatch = R.string.developer_options_mark_stories_completed
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 0,
+        targetViewId = R.id.mark_topics_completed_text_view,
+        stringIdToMatch = R.string.developer_options_mark_topics_completed
+      )
+    }
+  }
+
+  @Test
+  fun testDeveloperOptionsFragment_viewLogsIsDisplayed() {
+    launch<DeveloperOptionsActivity>(
+      createDeveloperOptionsActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      verifyItemDisplayedOnDeveloperOptionsListItem(
+        itemPosition = 1,
+        targetView = R.id.view_logs_text_view
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 1,
+        targetViewId = R.id.event_logs_text_view,
+        stringIdToMatch = R.string.developer_options_event_logs
+      )
+    }
+  }
+
+  @Test
+  fun testDeveloperOptionsFragment_overrideAppBehaviorsIsDisplayed() {
+    launch<DeveloperOptionsActivity>(
+      createDeveloperOptionsActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 2)
+      verifyItemDisplayedOnDeveloperOptionsListItem(
+        itemPosition = 2,
+        targetView = R.id.override_app_behaviors_text_view
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.show_all_hints_solution_text_view,
+        stringIdToMatch = R.string.developer_options_show_all_hints_solution
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.force_network_type_text_view,
+        stringIdToMatch = R.string.developer_options_force_network_type
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.force_crash_text_view,
+        stringIdToMatch = R.string.developer_options_force_crash
+      )
+    }
+  }
+
+  @Test
+  fun testDeveloperOptionsFragment_hintsAndSolutionSwitchIsUncheck() {
```

Comment:
can we write a test case to check click operation on this switch?

Reply:
Actually, the click operations are not implemented on any of the views in this PR. So, I think we cannot write tests to check click operation yet. WDYT?

Reply:
ok, could you add here which issue will be tracking the user interaction?

Reply:
Sure. will do that.

Reply:
Added TODO. PTAL. The issue is #3397.